Svc/DpCatalog/DpCatalog.cpp

```diff
@@ -534,7 +444,188 @@ Fw::CmdResponse DpCatalog::fillBinaryTree() {
 
 }  // end fillBinaryTree()
 
-bool DpCatalog::insertEntry(DpStateEntry& entry) {
+FwSizeType DpCatalog::determineDirectory(Fw::String fullFile) {
+    // Grab the directory string (up until the final slash)
+    // Could be found directly w/ a dirname func or regex
+    FwSignedSizeType loc = Fw::StringUtils::substring_find_last(
+        fullFile.toChar(), fullFile.length(), DIRECTORY_DELIMITER,
+        Fw::StringUtils::string_length(DIRECTORY_DELIMITER, sizeof(DIRECTORY_DELIMITER)));
+
+    // Seems like the logic works so long as the path styles match (i.e. relative vs absolute)
+    // Full path resolution might be a worthwhile add
+
+    // No directory delimiter found; return DP_MAX_DIRECTORIES to signal failure
+    if (-1 == loc) {
+        return DP_MAX_DIRECTORIES;
+    }
+
+    for (FwSizeType dir = 0; dir < this->m_numDirectories; dir++) {
+        const char* const dir_string = this->m_directories[dir].toChar();
+
+        // Compare both strings up to location of final slash
+        // memory safe since both are fixed width strings
+        // and loc is before the fixed width
+        if (strncmp(dir_string, fullFile.toChar(), static_cast<FwSizeType>(loc)) == 0) {
```

Comment:
We should use `StringUtils` instead of `strncmp` here if possible. If it's not possible, does `StringUtils` need to be updated?